test/logflare/lql/backend_transformer/clickhouse_test.exs

```diff
@@ -9,22 +9,37 @@ defmodule Logflare.Lql.BackendTransformer.ClickHouseTest do
   alias Logflare.Lql.Rules.SelectRule
 
   describe "behaviour implementation" do
-    test "dialect/0 returns 'clickhouse'" do
-      assert ClickHouse.dialect() == "clickhouse"
+    test "implements all required callbacks" do
+      assert function_exported?(ClickHouse, :transform_filter_rule, 2)
```

Comment:
BTW you can make this a bit more automated using `module.behaviour_info(:callbacks)` like this:

```elixir
iex> GenServer.behaviour_info(:callbacks)
[
  terminate: 2,
  init: 1,
  handle_info: 2,
  handle_continue: 2,
  handle_cast: 2,
  handle_call: 3,
  format_status: 2,
  format_status: 1,
  code_change: 3
]
```

Also using `@impl` makes the compiler check for missing callbacks AFAIK